src/task/index_task.rs

```diff
@@ -0,0 +1,109 @@
+use crate::cluster::{Cluster};
+use std::sync::Arc;
+use crate::errors::{ErrorKind, Result};
+use std::time::{Duration};
+use crate::task::{Status, Task};
+
+
+/// Struct for querying index creation status
+#[derive(Debug, Clone)]
+pub struct IndexTask {
+    cluster: Arc<Cluster>,
+    namespace: String,
+    index_name: String
+}
+
+
+static SUCCESS_PATTERN: &'static str = "load_pct=";
+static FAIL_PATTERN_201: &'static str = "FAIL:201";
+static FAIL_PATTERN_203: &'static str = "FAIL:203";
+static DELMITER: &'static str = ";";
+
+impl IndexTask {
+    /// Initializes IndexTask from client, creation should only be expose to Client
+    pub fn new(cluster: Arc<Cluster>, namespace: String, index_name: String) -> Self {
+        IndexTask {
+        	cluster: cluster,
+        	namespace: namespace,
+            index_name: index_name
+        }
+    }
+
+    fn build_command(namespace: String, index_name: String) -> String {
+        return format!("{}{}{}{}", "sindex/", namespace, "/", index_name);
+    }
+
+	fn parse_response(response: &str) -> Result<Status> {
+        match response.find(SUCCESS_PATTERN) {
+            None => {
+                match (response.find(FAIL_PATTERN_201), response.find(FAIL_PATTERN_203)) {
+                    (None, None) => bail!(ErrorKind::BadResponse(format!("Code 201 and 203 missing. Response: {}", response))),
+                    (_, _) => return Ok(Status::NotFound)
+                }
```

Comment:
We don't care about the index position of these patterns so we can use `str::contains` instead of `str::find` here:

```suggestion
                if response.contains(FAIL_PATTERN_201) || response.contains(FAIL_PATTERN_203)) {
                    return Ok(Status::NotFound)
                } else {
                    bail!(ErrorKind::BadResponse(format!("Code 201 and 203 missing. Response: {}", response))),
                }
```